src/components/editgrid/EditGrid.js

```diff
@@ -941,6 +941,11 @@ export default class EditGridComponent extends NestedArrayComponent {
       this.setNestedValue(component, editRow.data, flags);
     });
   }
+
+  resetValue() {
+    super.resetValue();
+    this.editRows = [];
```

Comment:
Now I think this will cause memory leaks. We should be destroying all components before setting this to empty. I think we need to change it to this.

```js
emptyRows() {
  this.editRows.forEach((editRow) => {
    editRow.components.forEach((comp) => {
      comp.destroy();
    });
  });
  this.editRows = [];
}
```

Then you will just call ```this.emptyRows();``` here.

Reply:
Actually it may need to be...

```js
emptyRows() {
  this.editRows.forEach((editRow, index) => this.destroyComponents(index));
  this.editRows = [];
}

```